src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -5159,6 +5177,24 @@ void Assembler::evpmadd52luq(XMMRegister dst, KRegister mask, XMMRegister src1,
   emit_int16((unsigned char)0xB4, (0xC0 | encode));
 }
 
+void Assembler::vpmadd52huq(XMMRegister dst, XMMRegister src1, Address src2, int vector_len) {
+  assert(vector_len == AVX_512bit ? VM_Version::supports_avx512ifma() : VM_Version::supports_avxifma(), "");
```

Comment:
What if vector length is 128 bit  and target does not support AVX_IFMA ?  AVX512_IFMA + AVX512_VL should still be still be sufficient to execute 52 bit MACs.